dashboard/app/controllers/registrations_controller.rb

```diff
@@ -72,8 +72,9 @@ def destroy
       }, status: :bad_request
       return
     end
-    TeacherMailer.delete_teacher_email(current_user).deliver_now if current_user.teacher?
-    destroy_dependent_users(current_user)
+    dependent_students = current_user.dependent_students
+    TeacherMailer.delete_teacher_email(current_user, dependent_students).deliver_now if current_user.teacher?
+    destroy_users(dependent_students << current_user)
```

Comment:
I wonder if we should not send the email until after we destroy the users, in case an exception occurs while destroying users.

Reply:
yeah, that's definitely safer. i'll swap line 76 and 77